wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/WirePlugin.kt

```diff
@@ -114,13 +114,16 @@ class WirePlugin : Plugin<Project> {
     }
 
     val targets = outputs.map { it.toTarget() }
-
     val wireTask = project.tasks.named("generateProtos") as TaskProvider<WireTask>
     wireTask.configure {
+      if (it.logger.isDebugEnabled) {
+        sourceInput.debug(it.logger)
+        protoInput.debug(it.logger)
+      }
       it.outputDirectories = outputs.map { output -> File(output.out!!) }
       it.source(sourceInput.configuration)
-      it.sourceInput = sourceInput
-      it.protoInput = protoInput
+      it.sourceInput = sourceInput.toLocations()
```

Comment:
This means the `toLocations` method is now called during configuration. Before, it was run during an `afterEvaluate` block which is pretty much the same. There might be a better way to organize things so that the dependency resolution would be performed at execution time, as it should but I don't know the plugin enough to refactor it this way.

Reply:
You could try make `WireTask.sourceInput` a `ListProperty<Location>` and wire a `Provider<Location>` to it here. Maybe changing `WireInput.toLocation()` to directly return the provider. This would defer dependency resolution to the task execution on a vintage build, and to the task graph calculation on a configuration-cache enabled build.

Reply:
I agree that it would be better, and I thought about using properties but all fields indeed should use lazy properties when it applies and it does for most of them. 
That would be a nice refresh for the wire plugin. Can it be done in a different PR though @eskatos ? My goal is not really to modernize the plugin but to have instant execution working.